### PR TITLE
OOP in ruby

### PR DESCRIPTION
**Pull request by tchasinga**
Implement classes and objects in Ruby.
Implement encapsulation and inheritance with Ruby.
Run a program using the command line.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    @id = id
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+  end
+
+  def of_age?
+    @age >= 18
+  end
+
```
Comment (Olamarx) [inline person.rb:18]: - [ ] As required, I will encourage you to make this method private.

File: student.rb [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# student.rb
+require_relative 'person'
+
+class Student < Person
+  def initialize(id, classroom)
+    super(id)
+    @classroom = classroom
+  end
```
Comment (Olamarx) [inline student.rb]: - [ ] There should be default parameters for age, name, and parent_permission but you are yet to add the parameter and the default data.

File: teacher.rb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# teacher.rb
+require_relative 'person'
+
+class Teacher < Person
+  def initialize(id, specialization)
+    super(id)
+    @specialization = specialization
+  end
+
+  def can_use_services?
```
Comment (Olamarx) [inline teacher.rb]: - [ ] Same thing applies to this.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @tchasinga, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Commit messages meaningful ✔️
- All methods in place ✔️

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — APPROVED** [review]
Hi @tchasinga,

## APPROVED 🟢

## Highlights
✔️ Project met all technical requirements

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**